File: 2019/go/main.go
package main

import (
	"fmt"
	"os"

	"github.com/mtslzr/advent-of-code/2019/go/day1"
)

func main() {
	if len(os.Args) < 2 {
		fmt.Println("Day not found.")
	} else {
		switch os.Args[1] {
		case "day1":
			day1.Part1()
			day1.Part2()
		case "day2":
			fmt.Println("Day 2!")
		default:
			fmt.Println("Day not found.")
		}
	}
}

File: 2019/rs/src/main.rs
extern crate clap;
mod day1;
use clap::{App, SubCommand};

fn main() {
  let matches = App::new("Advent of Code")
    .version("2019")
    .author("Matthew Salazar <m@tthewsalazar.com>")
    .subcommand(SubCommand::with_name("day1"))
    .get_matches();

  match matches.subcommand_name() {
    Some("day1") => {
      day1::part1();
      day1::part2();
    }
    None => println!("Day not found!"),
    _ => println!("Day not found!"),
  }
}
